src/lib/y2packager/storage_finder.rb

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "storage"
+
+module Y2Packager
+  # Mixing to find the right instance of Storage::Storage, i.e. the instance
+  # provided by Y2Storage if possible and a fallback with empty devicegraphs
+  # otherwise.
+  module StorageFinder
+    include Yast::Logger
+
+    # Storage object to use in order to access the current devicegraphs
+    #
+    # @return [Storage::Storage]
+    def storage_object
+      begin
+        require "y2storage"
+        Y2Storage::StorageManager.instance
+      rescue LoadError
```

Comment:
please elaborate a bit more, why it is valid use case to not have y2storage available. It is not much obvious for me.